src/client/debugger/jupyter/debuggingManager.ts

```diff
@@ -158,7 +172,32 @@ export class DebuggingManager implements IExtensionSingleActivationService, IDis
                                     this.commandManager,
                                     this.fs
                                 );
+                                this.disposables.push(
+                                    adapter.onDidSendMessage((msg: DebugProtocolMessage) => {
+                                        if ((msg as DebugProtocol.VariablesResponse).command === 'variables') {
+                                            this._onDidFireVariablesEvent.fire();
+                                        }
+                                        switch ((msg as DebugProtocol.ProtocolMessage).type) {
+                                            case 'request':
+                                                this.debugAdapterTrackers.forEach((d) => {
```

Comment:
Not sure why this necessary? Shouldn't debug adapter trackers just get called anyway?

Reply:
It doesn't get called :/
jupyterDebugService works the same way

Reply:
JupyterDebugService is different. It's the entire debugger for run by line for webviews. Because we couldn't switch VS code into debug mode.

Reply:
In your situation, the real debugger is there.